lxd/main_proxy.go

```diff
@@ -0,0 +1,164 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"os/signal"
+	"strconv"
+	"strings"
+	"syscall"
+
+	"github.com/lxc/lxd/shared"
+)
+
+func cmdProxyDevStart(args *Args) error {
```

Comment:
This function isn't needed, main.go already does all that error handling and return values stuff for you.

Just remove your cmdProxyDevStart function and rename run() to cmdProxyDevStart